pkg/query/statistics_optimizer_test.go

```diff
@@ -0,0 +1,512 @@
+package query
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+	"github.com/authzed/spicedb/pkg/schema/v2"
+	"github.com/authzed/spicedb/pkg/tuple"
+)
+
+func TestStatisticsOptimizer_ReorderUnion(t *testing.T) {
+	t.Parallel()
+
+	stats := DefaultStaticStatistics()
```

Comment:
why not use `subWithCost` here? so you can give different static selectivities to different iterators